test/lsptoolshost/integrationTests/codeactions.integration.test.ts

```diff
@@ -0,0 +1,349 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { describe, beforeAll, beforeEach, afterAll, test, expect, afterEach } from '@jest/globals';
+import testAssetWorkspace from './testAssets/testAssetWorkspace';
+import {
+    activateCSharpExtension,
+    closeAllEditorsAsync,
+    expectText,
+    openFileInWorkspaceAsync,
+} from './integrationHelpers';
+
+describe(`[${testAssetWorkspace.description}] Test Code Actions`, () => {
+    beforeAll(async () => {
+        await activateCSharpExtension();
+    });
+
+    beforeEach(async () => {
+        const fileName = path.join('src', 'app', 'CodeActions.cs');
+        await openFileInWorkspaceAsync(fileName);
+    });
+
+    afterAll(async () => {
+        await testAssetWorkspace.cleanupWorkspace();
+    });
+
+    afterEach(async () => {
+        await closeAllEditorsAsync();
+    });
+
+    test('Lightbulb displays actions', async () => {
+        const actions = await getCodeActions(new vscode.Range(0, 0, 0, 12));
+        expect(actions.length).toBeGreaterThanOrEqual(3);
+
+        // Verify we have unresolved code actions.
+        expect(actions[0].title).toBe('Remove unnecessary usings');
```

Comment:
consider using `actions.find(a => a.title === '')` unless ordering is something we want to bake into these tests.

Reply:
talked offline - ordering here is intentional; code actions should have a stable ordering in the list (can change based on client side changes, but not randomly).